src/lib/recursive_match.js

```diff
@@ -4,33 +4,40 @@
 function find_recursive(opener, closer, text) {
   var out = [];
   var last = [];
-  var chars = text.split('');
+  const chars = text.split('');
   var open = 0;
   for (var i = 0; i < chars.length; i++) {
+    const c = text[i];
     //increment open tag
-    if (chars[i] === opener) {
+    if (c === opener) {
       open += 1;
     }
     //decrement close tag
-    if (chars[i] === closer) {
+    else if (c === closer) {
       open -= 1;
       if (open < 0) {
         open = 0;
       }
+    } else if (last.length == 0) {
+      // If we're not inside of a pair of delimiters, we can discard the current letter.
+      // The return of this function is only used to extract images.
+      continue;
     }
-    if (open >= 0) {
```

Comment:
I think it's impossible for `open` to ever be less than zero since its only decremented in one place, and it's immediately clamped to 0 when it goes below.